assets/search.js

```diff
@@ -1,619 +1,666 @@
+    console.log("processing places");
+    let newObject = JSON.parse(localStorage.getItem("gTextData"));
+    newObject.eateries = request;
+    localStorage.setItem("gTextData", JSON.stringify(newObject));
+    populatePlaceResults();
+  }
 }
 
 // make cards for google results
 function populatePlaceResults() {
-    console.log("displaying places");
-    let placeArray = [];
-    let places = JSON.parse(localStorage.getItem("gTextData")).eateries;
-    console.log(places);
-    let displayEl = document.getElementById("google-results");
-    let index = 0;
-
-    places.forEach((place) => {
-        //create card
-        //add logic to not display closed businesses, but keep increasing the index of the array goes up
-        let cardEl = document.createElement("div");
-        cardEl.className = "card search-card";
-        cardEl.id = `card-${index}`;
-        displayEl.appendChild(cardEl);
-        //create array for cardEl.id and corresponding place_id
-        placeArray.push(place.place_id);
-        index++;
-
-        // div for image
-        let divEl = document.createElement("div");
-        divEl.className = "card-image";
-        cardEl.appendChild(divEl);
-
-        // image
-        let imgEl = document.createElement("img");
-        imgEl.className = "activator";
-        imgEl.setAttribute("src", "./assets/links/drinks.jpg");
-        divEl.appendChild(imgEl);
-
-        // add button
-        let buttonEl = document.createElement("button");
-        buttonEl.className =
-            "btn-floating halfway-fab waves-effect waves-light pink";
-        divEl.appendChild(buttonEl);
-        let iEl = document.createElement("i");
-        iEl.className = "material-icons bucketlist-add";
-        iEl.textContent = "favorite_border";
-        buttonEl.appendChild(iEl);
-
-        // create div for content
-        divEl = document.createElement("div");
-        cardEl.appendChild(divEl);
-        divEl.className = "card-content";
-
-        // place name
-        let spanEl = document.createElement("span");
-        spanEl.className = "card-title activator";
-        spanEl.textContent = place.name;
-        divEl.appendChild(spanEl);
-
-        // place address
-        pEl = document.createElement("p");
-        divEl.appendChild(pEl);
-        pEl.textContent = place.formatted_address;
-
-        // star rating
-        pEl = document.createElement("p");
-        divEl.appendChild(pEl);
-        var rating = Math.round(place.rating); //round rating to nearest integer
-        if (rating !== 0) {
-            for (var i = 1; i <= rating; i++) {
-                pEl.textContent += "✭";
-            }
-        } else {
-            pEl.textContent = "No rating";
+  console.log("displaying places");
+  let placeArray = [];
+  let places = JSON.parse(localStorage.getItem("gTextData")).eateries;
+  console.log(places);
+  let displayEl = document.getElementById("google-results");
+  let index = 0;
+  let randomNumCount = 0;
```

Comment:
index and randomNumCount here are redundant, you can use the index variable to index into your photo array, and just move the index++ to the line where you are increasing randomNumCount (or at the end ~line 388 if you might need the index for something else down the line